### PR TITLE
sectransp: EXC_BAD_ACCESS caused by non deterministic value

### PR DESCRIPTION
**Pull request by ebeworld**
We have following code in Curl.
...
 SecCertificateCopyCommonName(cert, &common_name);
#endif
          if(CFStringCompare(common_name, label_cf, 0) == kCFCompareEqualTo)

When the SecCertificateCopyCommonName function fails, it does not have to leave common_name in valid, state. So CFStringCompare is using the invalid result, hence causing EXC_BAD_ACCESS.

Fix is to check return value of the function.

### CODE REVIEW

File: lib/vtls/sectransp.c [hunk 1]
```diff
@@ -1090,9 +1090,9 @@ static OSStatus CopyIdentityWithLabel(char *label,
 #if CURL_BUILD_IOS
           common_name = SecCertificateCopySubjectSummary(cert);
 #elif CURL_BUILD_MAC_10_7
-          SecCertificateCopyCommonName(cert, &common_name);
+          OSStatus copy_status = SecCertificateCopyCommonName(cert, &common_name);
 #endif
-          if(CFStringCompare(common_name, label_cf, 0) == kCFCompareEqualTo) {
+          if(copy_status == errSecSuccess && CFStringCompare(common_name, label_cf, 0) == kCFCompareEqualTo) {
```
Comment (bagder) [inline lib/vtls/sectransp.c:1095]: `copy_status` is used here in the generic code while declared for macOS >= 10.7 only. The declaration needs to be moved, right?

Comment (ebeworld) [reply lib/vtls/sectransp.c:1095]: You are right. i was too fast. Fixing.

Comment (ebeworld) [reply lib/vtls/sectransp.c]: Done.

**Comment by bagder** [comment]
~~~
$ make checksrc
...
./vtls/sectransp.c:1167:103: warning: Longer than 79 columns (LONGLINE)
           if(copy_status == noErr && CFStringCompare(common_name, label_cf, 0) == kCFCompareEqualTo) {
checksrc: 0 errors and 1 warnings

**Review by bagder — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Pull request closed without merging**